client.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package cluster
+
+// KVStore provides access to the configuration store
+type KVStore interface {
+	// Get retrieves the value for the given key, marshalling it into the provided value
+	Get(key string, v interface{}) error
```

Comment:
If we are ok with deciding now that we will go the protobuf route we can make these of type `proto.Message`: https://godoc.org/github.com/golang/protobuf/proto#Message

Or we could make a new interface:
```go
type KVStoreValue interface {
  Marshall() ([]byte, error)
  Unmarshall(data []byte) error
}
```

We would have to create some container for protobufs though as `proto.Message` does not implement this.  We could:

```
type ProtobufValue interface {
  KVStoreValue
  Get() proto.Message
}

type protobufValue struct {
  value proto.Message
}

func NewProtobufValue(v proto.Message) {
  return &protobufValue{v}
}

func (v *protobufValue) Marshall() ([]byte, error) {
  return proto.Marshall(v.value)
}

func (v *protobufValue) Unmarshall(data []byte) error {
  return proto.Unmarshall(data, v.value)
}
```

Then use as such:
```
foo := NewThingProtobufObject() // type *ThingProtobufObject implements proto.Message
foo.innerKey = "innerValue"

kv := NewKVStore(...)

kv.Put("foo", NewProtobufValue(myProtobufValue))

bar := NewThingProtobufObject() // type *ThingProtobufObject implements proto.Message
err := kv.Get("bar", NewProtobufValue(bar))
// if err == nil then bar is returned value
```